acceptance/tests/openshift/basic_openshift_test.go

```diff
@@ -0,0 +1,163 @@
+	output, err := cmd.CombinedOutput()
+	require.NoErrorf(t, err, "failed to add hashicorp helm repo: %s", string(output))
+
+	// FUTURE for some reason NewHelmCluster creates a consul server pod that runs as root which
+	//   isn't allowed in OpenShift. In order to test OpenShift properly, we have to call helm and k8s
+	//   directly to bypass. Ideally we would just fix the framework that is running the pod as root.
+	cmd = exec.Command("kubectl", "create", "namespace", "consul")
+	output, err = cmd.CombinedOutput()
+	helpers.Cleanup(t, cfg.NoCleanupOnFailure, cfg.NoCleanup, func() {
+		cmd = exec.Command("kubectl", "delete", "namespace", "consul")
+		output, err = cmd.CombinedOutput()
+		assert.NoErrorf(t, err, "failed to delete namespace: %s", string(output))
+	})
+
+	require.NoErrorf(t, err, "failed to add hashicorp helm repo: %s", string(output))
+
+	cmd = exec.Command("kubectl", "create", "secret", "generic",
+		"consul-ent-license",
+		"--namespace", "consul",
+		`--from-literal=key=`+cfg.EnterpriseLicense)
+	output, err = cmd.CombinedOutput()
+	require.NoErrorf(t, err, "failed to add consul enterprise license: %s", string(output))
+
+	helpers.Cleanup(t, cfg.NoCleanupOnFailure, cfg.NoCleanup, func() {
+		cmd = exec.Command("kubectl", "delete", "secret", "consul-ent-license")
+		output, err = cmd.CombinedOutput()
+		assert.NoErrorf(t, err, "failed to delete secret: %s", string(output))
+	})
+
+	chartPath := "../../../charts/consul"
+	cmd = exec.Command("helm", "upgrade", "--install", "consul", chartPath,
+		"--namespace", "consul",
+		"--set", "global.name=consul",
+		"--set", "connectInject.enabled=true",
+		"--set", "connectInject.transparentProxy.defaultEnabled=false",
+		"--set", "connectInject.apiGateway.managedGatewayClass.mapPrivilegedContainerPorts=8000",
+		"--set", "global.acls.manageSystemACLs=true",
+		"--set", "global.tls.enabled=true",
+		"--set", "global.tls.enableAutoEncrypt=true",
+		"--set", "global.openshift.enabled=true",
+		"--set", "global.image="+cfg.ConsulImage,
+		"--set", "global.imageK8S="+cfg.ConsulK8SImage,
+		"--set", "global.imageConsulDataplane="+cfg.ConsulDataplaneImage,
+		"--set", "global.enterpriseLicense.secretName=consul-ent-license",
+		"--set", "global.enterpriseLicense.secretKey=key",
+	)
+	output, err = cmd.CombinedOutput()
+	helpers.Cleanup(t, cfg.NoCleanupOnFailure, cfg.NoCleanup, func() {
+		cmd := exec.Command("helm", "uninstall", "consul", "--namespace", "consul")
+		output, err := cmd.CombinedOutput()
+		require.NoErrorf(t, err, "failed to uninstall consul: %s", string(output))
+	})
+
+	require.NoErrorf(t, err, "failed to install consul: %s", string(output))
+
+	// this is normally called by the environment, but because we have to bypass we have to call it explicitly
+	logf.SetLogger(logr.New(nil))
+	logger.Log(t, "creating resources for OpenShift test")
+
+	cmd = exec.Command("kubectl", "apply", "-f", "../fixtures/cases/openshift/basic")
```

Comment:
We also typically use a helper to run kubectl commands:  https://github.com/hashicorp/consul-k8s/blob/main/acceptance/tests/partitions/partitions_connect_test.go#L219. I know you're bypassing the framework because of the root issue, but potentially you could still use the same pattern as other tests for these kubectl commands. Let me know if theres a reason not to though!!

Reply:
When I tried to use that function it got a bunch of errors from trying to connect to the static client pod, I'm not sure why